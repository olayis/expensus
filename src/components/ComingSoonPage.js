import React from 'react';
import { Link } from 'react-router-dom';
import programming from '../../public/img/programming.svg';

const ComingSoonPage = () => (
  <main className='main-content'>
    <div className='content-container'>
      <div className='error'>
        <img
          src={programming}
          alt='Programmer'
          className='error__illustration'
          width='1075'
          height='585'
        />
        <h2 className='error__text'>This page is coming soon.</h2>
        <div className='error__explanation'>
          Continue managing your expenses.
          <span className='error__action'>
            <Link className='btn btn--try-again' to='/'>
              Go Home
            </Link>
          </span>
        </div>
      </div>
    </div>
  </main>
);

export default ComingSoonPage;
