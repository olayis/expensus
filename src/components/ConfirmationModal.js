import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = (props) => {
  if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app');

  return (
    <div>
      <Modal
        contentLabel='Are you sure?'
        isOpen={!!props.deleteSelected}
        onRequestClose={props.stopDeleteDataSelected}
        closeTimeoutMS={200}
        className='modal'
        aria={{
          labelledby: 'heading',
          describedby: 'full_description',
        }}
      >
        <h3 id='heading' className='modal__title'>
          Delete Data
        </h3>
        <p id='full_description' className='modal__body'>
          Are you sure you want to delete all of your data on Expensus?
        </p>
        <button className='modal__btn--delete' onClick={props.handleDeleteData}>
          Delete
        </button>
        <button
          className='modal__btn--cancel'
          onClick={props.stopDeleteDataSelected}
        >
          Cancel
        </button>
      </Modal>
    </div>
  );
};

export default ConfirmationModal;
