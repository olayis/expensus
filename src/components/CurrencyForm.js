import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faExclamationCircle } from '@fortawesome/free-solid-svg-icons';

export default class CurrencyForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      currency: props.currency,
      currencyError: '',
    };
  }

  onCurrencyChange = (e) => {
    const currency = e.target.value;
    this.setState(() => ({ currency }));
  };

  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.currency) {
      this.setState(() => ({
        currencyError: 'Please select a valid currency',
      }));
    } else {
      this.setState(() => ({ currencyError: '' }));
      this.props.onSubmit({
        currency: this.state.currency.trim(),
      });
    }
  };

  render() {
    return (
      <form className='input-field' onSubmit={this.onSubmit}>
        <div className='input-field__wrapper'>
          <select
            className='dropdown dropdown--currency'
            value={this.state.currency}
            onChange={this.onCurrencyChange}
            aria-label='Select Currency'
            name='currency'
          >
            <option value=''>-- Select Currency --</option>
            <option value='Baht'>Baht (฿)</option>
            <option value='Danish Krone'>Danish Krone (DKK)</option>
            <option value='Dollar'>Dollar ($)</option>
            <option value='Dong'>Dong (₫)</option>
            <option value='Euro'>Euro (€)</option>
            <option value='Forint'>Forint (Ft)</option>
            <option value='Hryvnia'>Hryvnia (₴)</option>
            <option value='Koruna'>Koruna (Kč)</option>
            <option value='Lira'>Lira (₺)</option>
            <option value='Lev'>Lev (лв)</option>
            <option value='Naira'>Naira (₦)</option>
            <option value='Norwegian krone'>Norwegian krone (kr)</option>
            <option value='Pound Sterling'>Pound Sterling (£)</option>
            <option value='Rand'>Rand (R)</option>
            <option value='Real'>Real (R$)</option>
            <option value='Ruble'>Ruble (₽)</option>
            <option value='Swiss Franc'>Swiss Franc (CHF)</option>
            <option value='Yen'>Yen (¥)</option>
            <option value='Yuan'>Yuan (¥)</option>
            <option value='Zloty'>Złoty (PLN)</option>
          </select>
          {this.state.currencyError && (
            <span className='input-field__error-msg'>
              <FontAwesomeIcon className='fa-icon' icon={faExclamationCircle} />
              {this.state.currencyError}
            </span>
          )}
        </div>
        <button className='btn btn-currency'>Change Currency</button>
      </form>
    );
  }
}
