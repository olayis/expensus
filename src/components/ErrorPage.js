import React from 'react';
import { history } from '../routers/AppRouter';
import engineer from '../../public/img/QA-Engineer.svg';

const ErrorPage = () => (
  <div className='content-container'>
    <div className='error'>
      <img
        src={engineer}
        alt='A Quality Assurance Engineer'
        className='error__illustration'
        width='1075'
        height='585'
      />
      <h2 className='error__text'>Oops! Something went wrong.</h2>
      <div className='error__explanation'>
        Click here to try again.
        <span className='error__action'>
          <button className='btn btn--try-again' onClick={() => history.go(0)}>
            Try Again
          </button>
        </span>
      </div>
    </div>
  </div>
);

export default ErrorPage;
