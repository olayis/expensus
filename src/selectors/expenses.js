// Get Visible expenses
import moment from 'moment';

export default (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses
    .filter((expense) => {
      const createdAtMoment = moment(expense.createdAt);
      const startDateMatch = startDate
        ? startDate.isSameOrBefore(createdAtMoment, 'day')
        : true;
      const endDateMatch = endDate
        ? endDate.isSameOrAfter(createdAtMoment, 'day')
        : true;
      const textMatch = expense.description
        .toLowerCase()
        .includes(text.toLowerCase());

      return textMatch && startDateMatch && endDateMatch;
    })
    .sort((a, b) => {
      switch (sortBy) {
        case 'date(recent)':
          return a.createdAt < b.createdAt ? 1 : -1;

        case 'date(oldest)':
          return a.createdAt > b.createdAt ? 1 : -1;

        case 'amount(highest)':
          return a.amount < b.amount ? 1 : -1;

        case 'amount(lowest)':
          return a.amount > b.amount ? 1 : -1;

        case 'description(a_to_z)':
          return a.description.localeCompare(b.description);

        case 'description(z_to_a)':
          return b.description.localeCompare(a.description);

        case 'category':
          return a.category.localeCompare(b.category);
      }
    });
};
