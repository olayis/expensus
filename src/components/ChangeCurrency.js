import React from 'react';
import { connect } from 'react-redux';
import { startChangeLocale } from '../actions/locale';
import CurrencyForm from './CurrencyForm';

export class ChangeCurrency extends React.Component {
  onSubmit = ({ currency }) => {
    this.props.startChangeLocale({ currency });
    this.props.history.push('/');
    this.props.history.go(0);
  };

  goBack = () => {
    this.props.history.goBack();
  };

  render() {
    return (
      <main className='main-content'>
        <div className='content-container'>
          <div className='component-container component-container--currency'>
            <CurrencyForm
              onSubmit={this.onSubmit}
              currency={this.props.currency}
              goBack={this.goBack}
            />
          </div>
        </div>
      </main>
    );
  }
}

const mapStateToProps = (state) => ({
  currency: state.locale ? state.locale.currency : '',
});

const mapDispatchToProps = (dispatch) => ({
  startChangeLocale: (currency) => dispatch(startChangeLocale(currency)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChangeCurrency);
