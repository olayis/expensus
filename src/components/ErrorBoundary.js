import React from 'react';
import store from '../store/configureStore';
import ErrorPage from './ErrorPage';
import startLogError from '../actions/logError';

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    process.env.NODE_ENV === 'Production'
      ? store.dispatch(startLogError(error, errorInfo))
      : '';
  }

  render() {
    if (this.state.hasError) {
      return <ErrorPage />;
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
