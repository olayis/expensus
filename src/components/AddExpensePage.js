import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';
import { startChangeCurrency } from '../actions/locale';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.startChangeCurrency({ currency: 'Danish Krone' });
    this.props.history.push('/');
  };
  render() {
    return (
      <main className='main-content'>
        <div className='content-container'>
          <div className='add-expense-container'>
            <ExpenseForm onSubmit={this.onSubmit} />
          </div>
        </div>
      </main>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense)),
  startChangeCurrency: (currency) => dispatch(startChangeCurrency(currency)),
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
