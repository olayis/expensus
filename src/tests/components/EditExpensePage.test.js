import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { EditExpensePage } from '../../components/EditExpensePage';

let editExpense, startRemoveExpense, history, wrapper;

beforeEach(() => {
    editExpense = jest.fn();
    startRemoveExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(
        <EditExpensePage
            expense={expenses[1]}
            editExpense={editExpense}
            startRemoveExpense={startRemoveExpense}
            history={history}
        />
    );
});

test('should render editExpensePage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle editExpense on submit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(editExpense).toHaveBeenLastCalledWith(expenses[1].id, expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
});

test('should handle removeExpense on click', () => {
    wrapper.find('button').simulate('click');
    expect(startRemoveExpense).toHaveBeenLastCalledWith({
        id: expenses[1].id
    });
    expect(history.push).toHaveBeenLastCalledWith('/');
});