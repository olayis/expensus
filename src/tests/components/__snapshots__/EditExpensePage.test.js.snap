// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render editExpensePage correctly 1`] = `
<div>
  <h2>
    Edit Expense
  </h2>
  <ExpenseForm
    expense={
      Object {
        "amount": 35000,
        "createdAt": "1969-12-27T00:00:00.000Z",
        "description": "Rent",
        "id": "2",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
